### ISSUE TITLE
ValueError: cannot reshape array of size 90000 into shape (128,64)

### ISSUE THREAD

**Issue by err36** [eleurent/highway-env#181]
I seem to be having trouble when running this on my computer, I can run the code OK on colab. Do you know how I could fix this? I do not know why I am getting such a large array of 90,000. Thank you and apologies if this is a simple thing that I am missing.

import gym
import highway_env

env = gym.make('highway-v0')
from matplotlib import pyplot as plt
%matplotlib inline
config = {
   "observation": {
       "type": "GrayscaleObservation",
       "observation_shape": (128, 64),
       "stack_size": 4,
       "weights": [0.2989, 0.5870, 0.1140],  # weights for RGB conversion
       "scaling": 1.75,
   },
   "policy_frequency": 2
}
env.configure(config)
obs = env.reset()

_, axes = plt.subplots(ncols=4, figsize=(12, 5))
for i, ax in enumerate(axes.flat):
   ax.imshow(obs[i, ...].T, cmap=plt.get_cmap('gray'))
plt.show()

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-24-24fc44000b6a> in <module>
     13 }
     14 env.configure(config)
---> 15 obs = env.reset()
     16 
     17 _, axes = plt.subplots(ncols=4, figsize=(12, 5))

~/anaconda3/envs/ag/lib/python3.7/site-packages/highway_env/envs/common/abstract.py in reset(self)
    185         self._reset()
    186         self.define_spaces()  # Second, to link the obs and actions to the vehicles once the scene is created
--> 187         return self.observation_type.observe()
    188 
    189     def _reset(self) -> None:

~/anaconda3/envs/ag/lib/python3.7/site-packages/highway_env/envs/common/observation.py in observe(self)
     78     def observe(self) -> np.ndarray:
     79         new_obs = self._record_to_grayscale()
---> 80         new_obs = np.reshape(new_obs, self.observation_shape)
     81         self.state = np.roll(self.state, -1, axis=-1)
     82         self.state[:, :, -1] = new_obs

<__array_function__ internals> in reshape(*args, **kwargs)

~/.local/lib/python3.7/site-packages/numpy/core/fromnumeric.py in reshape(a, newshape, order)
    297            [5, 6]])
    298     """
--> 299     return _wrapfunc(a, 'reshape', newshape, order=order)
    300 
    301 

~/.local/lib/python3.7/site-packages/numpy/core/fromnumeric.py in _wrapfunc(obj, method, *args, **kwds)
     56 
     57     try:
---> 58         return bound(*args, **kwds)
     59     except TypeError:
     60         # A TypeError occurs if the object does have such a method in its

ValueError: cannot reshape array of size 90000 into shape (128,64)

**Comment by eleurent** [comment]
That is very strange... I just tried your code locally, and it works fine.

The problem seems to come from the https://github.com/eleurent/highway-env/blob/d6ef0d6d766c3b54a9f06cba3f09677d718109c3/highway_env/envs/common/observation.py#L95 function, which apparently returns an array of shape (90000,) instead of the intended (128, 64).

Maybe you could add a breakpoint / print statement to inspect the image returned by `self.viewer.get_image()`?
Something could be wrong with the pygame display... Does the environment renders correcly?

**Comment by err36** [comment]
Thank you for the quick reply! Really appreciate it. The problem seemed to be that I had a different version of the highway-env So when I checked the observation file you were referring to, I realised it was different from my one. 
My original install was done using: pip install highway-env
I therefore uninstalled the highway environment, deleted the package and reinstalled using: pip install git+https://github.com/eleurent/highway-env 
It works fine now.

I will close the issue.
